propulsion-reactor/Program.fs

```diff
@@ -48,6 +48,9 @@ module Args =
     let private defaultWithEnvVar varName argName = function
         | None -> getEnvVarForArgumentOrThrow varName argName
         | Some x -> x
+    let private defaultWithEnvVarOrFallback varName fallbackValue  = function
+        | Some x -> x
+        | None -> EnvVar.tryGet varName |> Option.defaultValue fallbackValue
```

Comment:
should this not be defaultWith - some of the stuff you chain to can throw

Reply:
If they do, we still want the application to fail. This function is expecting a value to fallback to when not found without failing. In this scenario we want it to fallback to already computed argument value, if it was not found application would've  throw an exception. This will also work when we want a default fallback value for an argument and do not want application to fail when not found.